### PR TITLE
add next adapter && state to handle it

### PR DESCRIPTION
**Pull request by SaraVieira**
This PR fixes the last issues we have with the lib in our project by adding a next adapter and also by triggering the callback on window search change

Also removed the need for dynamic import we had

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,9 +55,11 @@
     "typescript": "^4.5.5"
   },
   "dependencies": {
-    "@reach/dialog": "^0.16.2"
+    "@reach/dialog": "^0.16.2",
+    "zustand": "^3.6.9"
   },
   "peerDependencies": {
+    "next": "^12.0.10",
```
Comment (jfranciscosousa) [inline package.json:62]: We also need

```
"peerDependenciesMeta": {
  "next": {
    "optional": true
  }
}
```

Comment (SaraVieira) [reply package.json:62]: Done!

File: src/helpers.ts [hunk 2]
```diff
@@ -13,19 +21,31 @@ export const createURL = (urlParams: URLSearchParams) => {
...
-  const popStateEvent = new PopStateEvent('popstate', { state: null });
-  dispatchEvent(popStateEvent);
+const nextPush = async (href: string) => {
+  try {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const Router = await (await import('next/router')).default;
+    Router.push(href, undefined, { shallow: true });
+  } catch {
+    // we don't wanna do anything if we can't push
```
Comment (jfranciscosousa) [inline src/helpers.ts:31]: Can we just log the error at least?

**Review by jfranciscosousa — APPROVED** [review]

File: src/helpers.ts (hunk 2 above)
Comment (SaraVieira) [reply src/helpers.ts]: Done!

**Pull request merged**